### ISSUE TITLE
Inaccurate CWE for 8.3.1

### ISSUE THREAD

**Issue by jmeit-fwdsec** [OWASP/ASVS#1737]
The ASVS item **8.3.1** would be more accurately represented by [CWE-598:](https://cwe.mitre.org/data/definitions/598.html): _Use of GET Request Method With Sensitive Query Strings_, as opposed to its current mapping to [CWE-319](https://cwe.mitre.org/data/definitions/319.html): _Cleartext Transmission of Sensitive Information_.

**8.3.1 Description:** _Verify that sensitive data is only sent to the server in the HTTP message body or headers and that the URL and query string do not contain sensitive information, such as an API key or session token._

In other words, don't put sensitive data in the URL.

**Solution:** Verification 8.3.1 should be mapped to CWE-598.

https://github.com/OWASP/ASVS/blob/bd587a43701499602b427857fceba0137656dfb2/5.0/en/0x16-V8-Data-Protection.md?plain=1#L45

**Comment by elarlang** [comment]
CWE-319 is not correct, and CWE-598 is also not correct - it is about using HTTP GET method for _state changing_ actions, but not sending sensitive information in URL.

**Comment by elarlang** [comment]
With quick search I did not find anything suitable and the reason why we have not put too much effort on those values, is that we going to drop CWE mapping for v5.0, see https://github.com/OWASP/ASVS/issues/1481

My proposal at the moment we just delete the CWE value for 8.3.1 as it is not correct.

**Comment by jmeit-fwdsec** [comment]
Thank you for respond so quickly!
I don't see anything in the CWE-598 description that says it specifically relates to state changing actions.

**Description:**
"The web application uses the HTTP GET method to process a request and includes sensitive information in the query string of that request. "

**Extended Description:**
"The query string for the URL could be saved in the browser's history, passed through Referers to other web sites, stored in web logs, or otherwise recorded in other sources. If the query string contains sensitive information such as session identifiers, then attackers can use this information to launch further attacks."

**Comment by elarlang** [comment]
The title "Use of GET request methods" kind of says it directly.

**Comment by jmeit-fwdsec** [comment]
Perhaps we're interpreting that statement differently.
"Use of GET request methods" just means making GET requests, to me.

**Comment by elarlang** [comment]
GET is just one HTTP method.

The conflict comes when I'm using sensitive information as part of the URL but I'm using POST, PUT, HEAD or DELETE method. And it's not only about query strings, it's about entire URL.

**Comment by elarlang** [comment]
Anyway, thank you for opening the issue - current CWE value is clearly wrong. Proposed one is not correct, but for sure better than current one and if to read description 2 levels deep inside, it may serve to correct point for the requirement, but title limits it only to GET methods.

So now it's up to us to update CWE to 598 or to blank value.

**Comment by jmeit-fwdsec** [comment]
Ah! Ok. That makes sense to me. It doesn't fit because it _only_ covers GET.
Thank you for clarifying.
It wouldn't end up in the client's browser, but I suppose the full URLs could be logged on the server side.

**Comment by elarlang** [comment]
@tghosth - any opinions?

**Comment by tghosth** [comment]
I think 598 is closer than 319, even if it is not perfect. It was also the CWE for the requirements which were merged into this requirement. Where we have easy fixes, I am inclined to keep CWE as it may help us with future mappings...

**Comment by tghosth** [comment]
Opened #1742 to resolve
